lib/rack/attack/store_proxy.rb

```diff
@@ -22,9 +22,10 @@ def self.unwrap_active_support_stores(store)
 
         if store.instance_variable_defined?(:@data)
           client = store.instance_variable_get(:@data)
+          client_class = PoolProxy.unwrap_connection_pool_class(client)
```

Comment:
Bummer, I was expecting we would need less to actually support `ActiveSupport::Cache::MemCacheStore.new(pool_size: 2)` as a store.

Do we really need to unwrap the `MemCacheStore` to make the acceptance test green?
Do you know what is causing `MemCacheStore#increment` not work by default?

Reply:
If we fail to find a suitable Proxy class for the store, then Rack::Attack.cache.store is the unproxied `ActiveSupport::Cache::MemCacheStore` instance.

When we then call `store.increment` and `store.write` here: https://github.com/kickstarter/rack-attack/blob/master/lib/rack/attack/cache.rb#L57-L63

Normally our dalli proxy class would pass `raw: true` to the underlying instance, but because we don't have a proxy, a marshalled instance of `ActiveSupport::Cache::Entry` gets written to memcached.

The next call `increment` causes a dalli error of "Incr/decr on a non-numeric value".  This is caught by ActiveSupport's rescue handler (https://github.com/rails/rails/blob/4f09cff52c0ef771baac5f25c7c87a65068c0729/activesupport/lib/active_support/cache/mem_cache_store.rb#L202), and so returns nil

Reply:
Gotcha.

Thank you very much for the details!

So, if what is missing is only `raw: true`, maybe we can turn the acceptance test from red to green just by implementing a very small `MemCacheStoreProxy` with a `#write` method wrapper, analogous to the existing [`RedisCacheStoreProxy#write`](https://github.com/kickstarter/rack-attack/blob/03b8ce6f9e44dad2385aa04e5e6d67231a087296/lib/rack/attack/store_proxy/redis_cache_store_proxy.rb#L31-L33)?

As much as we can avoid messing with the internals of the original cache store the better.